### PR TITLE
Update samples pages for indexing and usability

### PR DESCRIPTION
**Pull request by wduraes**
This update adds tags to the samples so they can be automatically indexed by the Azure Samples page (https://docs.microsoft.com/en-us/samples/browse/ ). Also moving the Readme.md files closer to the sample code so that user can find both in one page, after searching on Azure Samples.

### CODE REVIEW

File: sdk/samples/iot/aziot_esp32/how_to_iot_hub_esp32.md [hunk 1]
```diff
@@ -1,3 +1,15 @@
+---
+page_type: sample
+description: Connecting an ESP32 device to Azure IoT using the Azure Embedded C SDK
```
Comment (ewertons) [inline sdk/samples/iot/aziot_esp32/how_to_iot_hub_esp32.md]:
```suggestion
description: Connecting an ESP32 device to Azure IoT using the Azure SDK for Embedded C
```

File: sdk/samples/iot/aziot_esp8266/how_to_iot_hub_esp8266_nodemcu.md [hunk 2]
```diff
@@ -1,3 +1,15 @@
+---
+page_type: sample
+description: Connecting an ESP8266 device to Azure IoT using the Azure Embedded C SDK
```
Comment (ewertons) [inline sdk/samples/iot/aziot_esp8266/how_to_iot_hub_esp8266_nodemcu.md]:
```suggestion
description: Connecting an ESP8266 device to Azure IoT using the Azure SDK for Embedded C
```

File: sdk/samples/iot/aziot_esp32/how_to_iot_hub_esp32.md (hunk 1 above)
Comment (ewertons) [reply sdk/samples/iot/aziot_esp32/how_to_iot_hub_esp32.md]: I like your better, but this is the title on the first page of the repo...

File: sdk/samples/iot/aziot_realtek_amebaD/how_to_iot_hub_realtek_amebaD.md [hunk 3]
```diff
@@ -1,3 +1,15 @@
+---
+page_type: sample
+description: Connecting a Realtek Ameba D device to Azure IoT using the Azure Embedded C SDK
```
Comment (ewertons) [inline sdk/samples/iot/aziot_realtek_amebaD/how_to_iot_hub_realtek_amebaD.md]:
```suggestion
description: Connecting a Realtek Ameba D device to Azure IoT using the Azure SDK for Embedded C
```

File: sdk/samples/iot/aziot_esp32/how_to_iot_hub_esp32.md [hunk 4]
```diff
@@ -1,3 +1,15 @@
+---
+page_type: sample
```
Comment (ewertons) [inline sdk/samples/iot/aziot_esp32/how_to_iot_hub_esp32.md]: Don't all these md files need to be renamed to readme.md ?

Comment (wduraes) [reply sdk/samples/iot/aziot_esp32/how_to_iot_hub_esp32.md]: sure thing. updated on the 2nd commit.

File: sdk/samples/iot/aziot_esp32/how_to_iot_hub_esp32.md (hunk 1 above)
Comment (wduraes) [reply sdk/samples/iot/aziot_esp32/how_to_iot_hub_esp32.md]: Agreed! Done in the 3rd commit.

File: sdk/samples/iot/aziot_esp8266/how_to_iot_hub_esp8266_nodemcu.md (hunk 2 above)
Comment (wduraes) [reply sdk/samples/iot/aziot_esp8266/how_to_iot_hub_esp8266_nodemcu.md]: Agreed! Done in the 3rd commit.

File: sdk/samples/iot/aziot_realtek_amebaD/how_to_iot_hub_realtek_amebaD.md (hunk 3 above)
Comment (wduraes) [reply sdk/samples/iot/aziot_realtek_amebaD/how_to_iot_hub_realtek_amebaD.md]: Agreed! Done in the 3rd commit.

**Review by ewertons — APPROVED** [review]

File: sdk/docs/iot/README.md [hunk 5]
```diff
@@ -1,226 +1,773 @@
-# Azure IoT Clients
+# Azure IoT Samples
```
Comment (danewalton) [inline sdk/docs/iot/README.md:1]: I don't think I agree with this change. This is the first README in `/sdk/docs/iot/` and its title is `Azure IoT Samples`. I think this should still have an intro which explains the sdk, building, etc. It's a catchall that then can feed people into specific things like tests, API docs, samples, etc.

**Review by danewalton — CHANGES_REQUESTED** [review]

File: sdk/docs/iot/README.md (hunk 5 above)
Comment (wduraes) [reply sdk/docs/iot/README.md]: last commit solved this.

**Review by danewalton — APPROVED** [review]
looks good now

**Pull request merged**